File: after-ioi/test/grader/mod_good.pas
{
TASK: mod
LANG: PASCAL
}

var
    s,u,n,r:integer;
begin
    readln(s,u);
    r := 1;
    repeat
        n := (s + u) >> 1 ;
        writeln(n);
        readln(r);
        case r of
            -1 : u := n - 1;
             1 : s := n + 1;
        end;
    until (r = 0);
end.

File: after-ioi/test/grader/mod_good.cpp
/*
TASK: mod
LANG: C++
*/

#include<iostream>

using namespace std;

int main()
{   int s,u,n,r;
    cin >> s >> u;
    while(1)
    {   n = (s+u)/2;
        cout << n << endl;
        cin >> r;
        if (r<0)
            u = n-1;
        else if (r>0)
            s = n+1;
        if (r==0)
            break;
    } 

    return 0;
}
